doc/doxygen/Doxyfile.in

```diff
@@ -754,7 +754,7 @@ WARN_LOGFILE           =
 # spaces.
 # Note: If this tag is empty the current directory is searched.
 
-INPUT                  = @CMAKE_CURRENT_LIST_DIR@/..
+INPUT                  = @CMAKE_CURRENT_LIST_DIR@/../..
```

Comment:
Could you keep CMakeLists.txt in doc? Then you can maybe make the CMakeLists produce the userdoc as well.